src/coreclr/vm/loaderallocator.hpp

```diff
@@ -941,6 +952,36 @@ class AssemblyLoaderAllocator : public LoaderAllocator
 
 typedef VPTR(AssemblyLoaderAllocator) PTR_AssemblyLoaderAllocator;
 
+#ifndef DACCESS_COMPILE
+class CollectibleMoveableGCPointerTracker
```

Comment:
I was trying quite hard to avoid perturbing the GCInterface as I understand that is quite difficult for the GC team to handle due to the preference for having the GCInterface stable across multiple versions of the runtime. However, if @Maoni0 would prefer for me to write a new handle type, I can do that instead.

Reply:
> I understand that is quite difficult for the GC team to handle due to the preference for having the GCInterface stable across multiple versions of the runtime

Adding new things is not a problem.

Removing or changing existing things is a problem for standalone GC build that the GC team wants to be usable on old runtimes. For this reason, the standalone GC build includes extra cruft that is no longer used: https://github.com/dotnet/runtime/blob/788f5726850d5bd85141f821f5329cb6e5a80a18/src/coreclr/gc/handletableconstants.h#L10-L13

Reply:
I think a new handle type would be good here.

Reply:
I've defined a new handle type now. I've called it HNDTYPE_WEAK_INTERIOR_POINTER. Would you like for me to try and split off the definition of this new handle type into a separate PR?